x-pack/plugins/security_solution/public/common/hooks/endpoint/upgrade.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { useEffect } from 'react';
+import { HttpFetchOptions, HttpStart } from 'src/core/public';
+import { useKibana } from '../../../../../../../src/plugins/kibana_react/public';
+import {
+  epmRouteService,
+  appRoutesService,
+  CheckPermissionsResponse,
+  BulkInstallPackagesResponse,
+} from '../../../../../ingest_manager/common';
+import { StartServices } from '../../../types';
+import { useIngestEnabledCheck } from './ingest_enabled';
+
+/**
+ * Requests that the endpoint package be upgraded to the latest version
+ *
+ * @param http an http client for sending the request
+ * @param options an object containing options for the request
+ */
+const sendUpgradeEndpointPackage = async (
+  http: HttpStart,
+  options: HttpFetchOptions = {}
+): Promise<BulkInstallPackagesResponse> => {
+  return http.post<BulkInstallPackagesResponse>(epmRouteService.getBulkInstallPath(), {
+    ...options,
+    body: JSON.stringify({
+      packages: ['endpoint'],
+    }),
+  });
+};
+
+/**
+ * Checks with the ingest manager if the current user making these requests has the right permissions
+ * to install the endpoint package.
+ *
+ * @param http an http client for sending the request
+ * @param options an object containing options for the request
+ */
+const sendCheckPermissions = async (
+  http: HttpStart,
+  options: HttpFetchOptions = {}
+): Promise<CheckPermissionsResponse> => {
+  return http.get<CheckPermissionsResponse>(appRoutesService.getCheckPermissionsPath(), {
+    ...options,
+  });
+};
+
+export const useUpgradeEndpointPackage = () => {
+  const context = useKibana<StartServices>();
+  const { allEnabled: ingestEnabled } = useIngestEnabledCheck();
+
+  useEffect(() => {
+    const abortController = new AbortController();
+
+    // cancel any ongoing requests
+    const cleanup = () => {
```

Comment:
❔ `cleanup` as a name suggests this should _always_ be done, which is a little confusing because this sends an `abort` through the connection. Rename or clarify, perhaps?